### PR TITLE
minor changes to docker and token_bucket_rate_limiter

### PR DESCRIPTION
**Pull request by junjieqian**
1. Change the Golang version in comments from 1.8.1 to 1.8.2.
2. Update argument check in start.sh.
3. Change the timestamp `now` in `TryConsume` in token bucket from entering the function to acquiring the lock. This is to eliminate the waiting time for the lock.

### CODE REVIEW

File: docker/start.sh [hunk 1]
```diff
@@ -73,6 +73,11 @@ init_env() {
     fi
 }
 
+if [ $# -eq 0 ]; then
+    echo 'CADENCE_HOME required.'
+    exit 0
```
Comment (venkat1109) [inline docker/start.sh]: exit with non-zero return value

File: docker/start.sh [hunk 2]
```diff
@@ -73,6 +73,11 @@ init_env() {
     fi
 }
 
+if [ $# -eq 0 ]; then
+    echo 'CADENCE_HOME required.'
```
Comment (venkat1109) [inline docker/start.sh]: update this message to be more meaningful i.e. something like below
```
Missing required argument. See usage below:
Usage: ./start.sh <path_to_cadence_repo>
```

File: docker/Dockerfile [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 FROM debian:jessie
 
-# get golang 1.8.1
+# get golang 1.8.2
```
Comment (junjieqian) [inline docker/Dockerfile]: Thanks for the comments! The new changes addressed the comments.

**Comment by venkat1109** [comment]
This PR is out of date. We have a dependency on golang/x/ratelimiter now and the Dockerfile has also changed since then. So abandoning this.

**Pull request closed without merging**